core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -129,6 +136,8 @@ public void initialize(String name, Map<String, String> properties) {
     this.fs = Util.getFs(new Path(warehouseLocation), conf);
 
     String fileIOImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
+    this.suppressIOException = Boolean.parseBoolean(properties.get(CatalogProperties.HADOOP_SUPPRESS_IOEXCEPTION));
+
```

Comment:
Can you move this to after `fileIO` is set? The line above and that line are related so we shouldn't separate them.